### ISSUE TITLE
Crash On EditText Click

### ISSUE THREAD

**Issue by Esseme** [mukeshsolanki/country-picker-android#42]
`java.lang.NoSuchFieldError: No static field country_flag of type I in class Lcom/mukesh/countrypicker/R$id; or its superclasses (declaration of 'com.mukesh.countrypicker.R$id' appears in /data/app/com.imarad.app-2/base.apk)
                                                                  at com.mukesh.countrypicker.CountriesAdapter$ViewHolder.<init>(CountriesAdapter.java:67)
                                                                  at com.mukesh.countrypicker.CountriesAdapter.onCreateViewHolder(CountriesAdapter.java:36)
                                                                  at com.mukesh.countrypicker.CountriesAdapter.onCreateViewHolder(CountriesAdapter.java:14)
                                                                  at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:6519)
                                                                  at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5706)
                                                                  at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5589)
                                                                  at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5585)
                                                                  at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2231)
                                                                  at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1558)
                                                                  at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1518)
                                                                  at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:610)
                                                                  at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3719)
                                                                  at android.support.v7.widget.RecyclerView.onMeasure(RecyclerView.java:3135)
                                                                  at android.view.View.measure(View.java)
                                                                  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java)
                                                                  at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java)
                                                                  at android.widget.LinearLayout.measureVertical(LinearLayout.java)
                                                                  at android.widget.LinearLayout.onMeasure(LinearLayout.java)
                                                                  at android.view.View.measure(View.java)
                                                                  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java)
                                                                  at android.widget.FrameLayout.onMeasure(FrameLayout.java)
                                                                  at android.view.View.measure(View.java)
                                                                  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java)
                                                                  at android.widget.FrameLayout.onMeasure(FrameLayout.java)
                                                                  at android.view.View.measure(View.java)
                                                                  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java)
                                                                  at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java)
                                                                  at android.widget.LinearLayout.measureVertical(LinearLayout.java)
                                                                  at android.widget.LinearLayout.onMeasure(LinearLayout.java)
                                                                  at android.view.View.measure(View.java)
                                                                  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java)
                                                                  at android.widget.FrameLayout.onMeasure(FrameLayout.java)
                                                                  at com.android.internal.policy.impl.PhoneWindow$DecorView.onMeasure(PhoneWindow.java)
                                                                  at android.view.View.measure(View.java)
                                                                  at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java)
                                                                  at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java)
                                                                  at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java)
                                                                  at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java)
                                                                  at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java)
                                                                  at android.view.Choreographer$CallbackRecord.run(Choreographer.java)
                                                                  at android.view.Choreographer.doCallbacks(Choreographer.java)
                                                                  at android.view.Choreographer.doFrame(Choreographer.java)
                                                                  at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java)
                                                                  at android.os.Handler.handleCallback(Handler.java)
                                                                  at android.os.Handler.dispatchMessage(Handler.java)
                                                                  at android.os.Looper.loop(Looper.java)
                                                                  at android.app.ActivityThread.main(ActivityThread.java)
                                                                  at java.lang.reflect.Method.invoke(Native Method)
                                                                  at java.lang.reflect.Method.invoke(Method.java:372)
                                                                  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java)
                                                                  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java)
`
I am using **Version 1.2.2** of the library.
Basically, I am doing an `onClick` on the EditText but I persistently get this error

**Comment by mukeshsolanki** [comment]
can you share a snippet of your code so i can try to reproduce this issue

**Comment by Esseme** [comment]
`public class SignUpActivity extends AppCompatActivity {
CountryPicker countryPicker;
@BindView(R.id.countryEditText) EditText countryEdit;

@Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
countryPicker();
}

//Country Picker Method
    private void countryPicker(){
        countryPicker = new CountryPicker.Builder().with(SignUpActivity.this).listener(new OnCountryPickerListener() {
            @Override
            public void onSelectCountry(Country country) {
                countryEdit.setText(country.getName());

            }
        }).build();

        countryEdit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                countryPicker.showDialog(getSupportFragmentManager());
            }
        });

    }
}`

**Comment by mukeshsolanki** [comment]
@EssemeI can't seem to reproduce the issue with the above code. also, I see your `onCreate` method is missing a few lines

```@Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_signup);
    ButterKnife.bind(this);
countryPicker();
}```

**Comment by mukeshsolanki** [comment]
then this would not be related to the library and more on your implementation of the edit text.

**Comment by Esseme** [comment]
Cool, I am going to investigate!

**Comment by mukeshsolanki** [comment]
@Esseme any update on this?

**Comment by AndreyGrowl** [comment]
Got the same error =\
For some reason it just stopped working.

I'm using 2.0.1 version.

**Comment by BesartLaci** [comment]
### Replicated Behavior  
using compile 'com.github.mukeshsolanki:country-picker-android:2.0.1'

I m using the CountryPicker for an EditText onClick Event.

**Gradle Build Properties**
```setting  
android {
    compileSdkVersion 27
    buildToolsVersion '25.0.3'
    defaultConfig {
        minSdkVersion 16
        targetSdkVersion 27
        multiDexEnabled true

        vectorDrawables.useSupportLibrary = true
    }
```
**Class __RegistrationFragment**
```java 
    @Nullable
    @Override
public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment__, container, false);
try {
.
.
.
           selectedCountry = view.findViewById(R.id.et_country);

            selectedCountry.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {

                    CountryPicker.Builder builder = new CountryPicker.Builder()
                            .with(getContext())
                            .listener(new OnCountryPickerListener() {
                                @Override
                                public void onSelectCountry(Country country) {
                                    selectedCountry.setText(country.getCode());
                                }
                            });
                    CountryPicker picker = builder.build();
                    picker.showDialog(getActivity());


                }
            });  
.
.
.

  }
```  

**getAcitivity()**
```java  
@Nullable
    public final FragmentActivity getActivity() {
        return this.mHost == null ? null : (FragmentActivity)this.mHost.getActivity();
    }  
```

**Terminal Error**

```terminal  
 java.lang.NoSuchFieldError: No static field country_flag of type I in class Lcom/mukesh/countrypicker/R$id; or its superclasses (declaration of 'com.mukesh.countrypicker.R$id' appears in /data/app/com.karatbars.app-Yz8foh0n-DxQsKST-1jw2g==/split_lib_slice_0_apk.apk)
        at com.mukesh.countrypicker.CountriesAdapter$ViewHolder.<init>(CountriesAdapter.java:72)
        at com.mukesh.countrypicker.CountriesAdapter.onCreateViewHolder(CountriesAdapter.java:40)
        at com.mukesh.countrypicker.CountriesAdapter.onCreateViewHolder(CountriesAdapter.java:16)
        at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:6685)
        at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5869)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5752)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5748)
        at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2232)
        at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1559)
        at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1519)
        at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:614)
        at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3812)
        at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3529)
        at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:4082)
        at android.view.View.layout(View.java:20960)
        at android.view.ViewGroup.layout(ViewGroup.java:6279)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1842)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1686)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1595)
        at android.view.View.layout(View.java:20960)
        at android.view.ViewGroup.layout(ViewGroup.java:6279)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at android.view.View.layout(View.java:20960)
        at android.view.ViewGroup.layout(ViewGroup.java:6279)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at android.view.View.layout(View.java:20960)
        at android.view.ViewGroup.layout(ViewGroup.java:6279)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at com.android.internal.policy.DecorView.onLayout(DecorView.java:926)
        at android.view.View.layout(View.java:20960)
        at android.view.ViewGroup.layout(ViewGroup.java:6279)
        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3037)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2545)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1636)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7946)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1092)
        at android.view.Choreographer.doCallbacks(Choreographer.java:893)
        at android.view.Choreographer.doFrame(Choreographer.java:812)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1078)
        at android.os.Handler.handleCallback(Handler.java:907)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:216)
        at android.app.ActivityThread.main(ActivityThread.java:7593)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
```
**Debug CountryPicker.showDialog()**
![grafik](https://user-images.githubusercontent.com/32385936/58539488-bf6ce200-81f7-11e9-92e8-2f39ab78aab4.png)


- the App crashes when i click the selectedCountry EditText field
- the CountryPicker dialog doesn´t pop up

**maybe you can give me a hint if i m using your library the wrong way?**


you mentiont in [your commit](https://github.com/mukeshsolanki/country-picker-android/issues/42#issuecomment-382270267) also to use  
```java  
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_signup);
    ButterKnife.bind(this);
countryPicker();
}  
```
but i dont realy understand where to do this?
i used the "how to" from https://medium.com/@themickyvirus/how-to-add-a-country-picker-for-android-in-few-lines-5503856f8fce

**Comment by matt39** [comment]
Same problem here, did anybody found a solution?

**Comment by matt39** [comment]
I found my problem.
I'm also using [android-phone-field](https://github.com/lamudi-gmbh/android-phone-field) and that lib has a layout file named item_country.xml that conflicts with a layout file of country-picker-android with the same name.

Does anybody know how to solve?

**Comment by oluwabajio** [comment]
i am having this same issue.

 java.lang.NoSuchFieldError: No field countries_recycler_view of type I in class Lcom/mukesh/countrypicker/R$id; or its superclasses (declaration of 'com.mukesh.countrypicker.R$id' appears in /data/app/tingtel.android-2/base.apk)
        at com.mukesh.countrypicker.CountryPicker.initiateUi(CountryPicker.java:535)
        at com.mukesh.countrypicker.CountryPicker.showDialog(CountryPicker.java:385)
        at tingtel.android.fragments.NetworkFragment$4.onClick(NetworkFragment.java:110)
